frontend/public/components/catalog.jsx

```diff
@@ -0,0 +1,209 @@
+    const imagestreamsItems = this.normalizeImagestreams(imagestreams.data, 'ImageStream');
+
+    const items = [
+      ...clusterServiceClassItems,
+      ...imagestreamsItems
+    ];
+
+    const filteredItems = this.filterItems(items);
+    this.setState({items, filteredItems});
+  }
+
+  normalizeClusterServiceClasses(serviceClasses, kind) {
+    return _.map(serviceClasses, serviceClass => {
+      const tileName = serviceClassDisplayName(serviceClass);
+      const tileIcon = <ClusterServiceClassIcon serviceClass={serviceClass} />;
+      const tileDescription = _.get(serviceClass, 'spec.description');
+      const tileVendor = _.get(serviceClass, 'spec.externalMetadata.providerDisplayName');
+
+      return {
+        resource: serviceClass,
+        kind,
+        tileName,
+        tileIcon,
+        tileDescription,
+        tileVendor
+      };
+    });
+  }
+
+  normalizeImagestreams(imagestreams, kind) {
+    imagestreams = _.map(imagestreams, imagestream => {
+      // A valid specTag has a 'builder' tag. no 'hidden' tag, and exists in 'status.tags'.
+      const builderSpecTagName = getBuilderSpecTagName(imagestream);
+      if (builderSpecTagName) {
+        const tileName = _.get(imagestream, ['metadata', 'annotations', 'openshift.io/provider-display-name']) || imagestream.metadata.name;
+        // TODO: icon, description and vendor should be based on selected version?
+        const tileIcon = <span
+          className={`icon ${normalizeIconClass(getImageStreamTagAnnotation(imagestream, 'iconClass', builderSpecTagName))}`}
+          aria-hidden="true">
+        </span>;
+        const tileDescription = getImageStreamTagAnnotation(imagestream, 'description', builderSpecTagName);
+        const tileVendor = getImageStreamTagAnnotation(imagestream, 'openshift.io/provider-display-name', builderSpecTagName);
+        return {
+          resource: imagestream,
+          kind,
+          tileName,
+          tileIcon,
+          tileDescription,
+          tileVendor
+        };
+      }
+      return {
+        hidden: true
+      };
+    });
+
+    return _.filter(imagestreams, imagestream => {
+      return !imagestream.hidden;
+    });
+  }
+
+  handleFilterChange(event) {
+    this.setState({filterValue: event.target.value});
+  }
+
+  clearFilter() {
+    this.setState({filterValue: ''});
+  }
+
+  render() {
+    const {loaded, loadError} = this.props;
+    const {filteredItems} = this.state;
+
+    return <div className="co-m-pane">
+      <div className="co-m-pane__body">
+        <StatusBox data={filteredItems} loaded={loaded} loadError={loadError} label="Resources">
+          <CatalogList items={filteredItems} />
+        </StatusBox>
+      </div>
+    </div>;
+  }
+}
+
+CatalogListPage.displayName = 'CatalogList';
+
+CatalogListPage.propTypes = {
+  obj: PropTypes.object
+};
+
+//{namespace}
+export const Catalog = () => {
+  const resources = [
+    {
+      isList: true,
+      kind: 'ClusterServiceClass',
```

Comment:
Need to check if service catalog is enabled before requesting service catalog items